plugins/hashsum_download/server/__init__.py

```diff
@@ -118,7 +129,46 @@ def _getFirstFileByHash(self, algo, hash, user=None):
         return None
 
 
+def _computeHashHook(event):
+    """
+    Computes all supported checksums on a given file. Downloads the
+    file data and stream-computes all required hashes on it, saving
+    the results in the file document.
+    """
+    if ModelImporter.model('setting').get(PluginSettings.AUTO_COMPUTE, default=False):
+        _computeHash(event.info['file'])
+
+
+def _computeHash(file):
+    toCompute = SUPPORTED_ALGORITHMS - set(file)
+    toCompute = {alg: getattr(hashlib, alg)() for alg in toCompute}
+
+    if not toCompute:
+        return
+
+    fileModel = ModelImporter.model('file')
+    with fileModel.open(file) as fh:
+        while True:
+            chunk = fh.read(65536)
+            if not chunk:
+                break
+            for hash in six.viewvalues(toCompute):
```

Comment:
nit: can you use a different name here for `hash`, because elsewhere in this module `hash` is a var name for the actual hash digest value.  maybe `hasher` or `hashComputation` or `hashCalculation`, `ongoing`, `digester`?